tensorflow_datasets/scripts/refactoring/refactoring_dataset.py

```diff
@@ -0,0 +1,148 @@
+from absl import app
+from dataclasses import dataclass
+import os
+import inspect
+import pathlib
+import posixpath
+import shutil
+import tensorflow_datasets as tfds
+from tensorflow_datasets.core import naming
+from typing import Type
+
+"""A refactoring script to aggregate datasets file from diferent directories
+
+my_dataset/
+    fake_data/
+    __init__.py
+    dataset.py
+    dataset_test.py
+    make_fake_data.py
+    checksum.txt
+
+To run this script
+python3 -m tensorflow_datasets.scripts.refactoring.refactoring_dataset
+
+New datasets will be generated in tensorflow_datasets/refactored_dataset
+
+"""
+
+NEW_TFDS_DIR = 'tensorflow_datasets/refactored_dataset'
+URL_CHECKSUM_DIR = 'tensorflow_datasets/url_checksums'
+FAKE_DATA_DIR = 'tensorflow_datasets/testing/test_data/fake_examples'
+FAKE_DATA_SCRIPT_DIR = 'tensorflow_datasets/testing/fake_data_generation'
+
+_INIT_FILE = """\
+\"""{dataset_name} dataset.\"""
+from tensorflow_datasets.{dataset_type}.{dataset_name}.{dataset_name} import {dataset_name_cls}
+"""
+
+
+@dataclass
+class BuilderInfo:
+  '''Dataset metadata
+  dataset_file: path to dataset file
+  dataset_name: name of the dataset
+  dataset_type: type of the dataset
+  dataset_path: path to dataset type directory
+  '''
+  dataset_file: pathlib.Path
+  dataset_name: str
+  dataset_type: str
+  dataset_path: pathlib.Path
+
+
+def create_dirs(dataset_path: str) -> None:
```

Comment:
Please use pathlib here and bellow.
You shouldn't use open(), os.path.join,... but instead the pathlib equivalent.

Reply:
Done